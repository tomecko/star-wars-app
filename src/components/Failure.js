import React from 'react';
import { Button, Card, CardBody } from 'reactstrap';

export function Failure({ error, send }) {
  return (
    <Card>
      <CardBody>
        <p>
          Something went wrong
          {error ? ` (details: ${error.message})` : null}.
        </p>
        <Button
          color="danger"
          onClick={() => send('RETRY')}
        >
          let's try again!
        </Button>
      </CardBody>
    </Card>
  );
};
