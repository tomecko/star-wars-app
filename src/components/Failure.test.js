import React from 'react';
import renderer from 'react-test-renderer';

import { Failure } from './Failure';

test('matches snapshot', () => {
  const component = renderer.create(
    <Failure />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
