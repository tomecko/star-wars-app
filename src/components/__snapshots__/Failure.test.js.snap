// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <p>
      Something went wrong
      .
    </p>
    <button
      aria-label={null}
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      let's try again!
    </button>
  </div>
</div>
`;
